daemon/container.go

```diff
@@ -896,21 +896,21 @@ func (container *Container) Copy(resource string) (io.ReadCloser, error) {
 		return nil, err
 	}
 
-	basePath, err := container.getResourcePath(resource)
-	if err != nil {
+	if err := container.mountVolumes(); err != nil {
```

Comment:
I notice that we're calling
```go
container.unmountVolumes()
container.unmount()
```

in every error check for the rest of the function. How about we do defer with a condition on the returned err instead?

```go
func (container *Container) Copy(resource string) (r io.ReadCloser, err error) {
	defer func() {
		if err != nil {
			container.Unmount()
		}
	}

 	if err := container.Mount(); err != nil {
 		return nil, err
 	}

 	defer func() {
 		if err != nil {
 			container.unmountVolumes()
 		}
 	}

 	if err := container.mountVolumes(); err != nil {
 		return nil, err
 	}

 	...
```

Then if the call returns with a non-nil error, `unmountVolumes()` and `Unmount()` will be called on this container without the need to remember it each time we need to return early.